src/future.cpp

```diff
@@ -240,6 +240,14 @@ namespace detail {
     good since we need a concrete interface at this level. Above code could be simpler.
 */
 
+void async_(std::chrono::milliseconds delay, function<void()> f) {
```

Comment:
Delay can be ambiguous - better to treat this as getting executed at a specific time.